airflow/operators/python_operator.py

```diff
@@ -330,13 +330,28 @@ def _write_string_args(self, filename):
 
     def _write_args(self, input_filename):
         # serialize args to file
+        if self.use_dill:
+            serializer = dill
+        else:
+            serializer = pickle
+        # some args from context can't be loaded in virtual env
+        invalid_args = set(['dag', 'task', 'ti'])
         if self._pass_op_args():
+            kwargs = {}
+            for key, value in self.op_kwargs.items():
```

Comment:
More from a code point of view, but I would do something like:
```python
MacBook-Pro-van-Fokko:home-analytics fokkodriesprong$ python2
Python 2.7.17 (default, Dec 23 2019, 21:25:33) 
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.16)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> invalid_args = {'dag', 'task', 'ti'}
>>> op_kwargs = {
...   'ti': 'Ignored',
...   'foo': 'bar'
... }
>>> 
>>> def serialize(self, item):
...     try:
...         return serializer.loads(serializer.dumps(value))
...     except Exception as e:
...         msg = """
...         "Exception %s found while serializing argument
...         object: %s on op_kwargs key %s ...skipping..."
...         """ % (e, value, key)
...         self.log.debug(msg)
... 
>>> # Replace it with something simple for the sake of showing how this works.
>>> def serialize(item):
...     return item
... 
>>> kwargs = {key: serialize(item) for key, item in op_kwargs.items() if key not in invalid_args}
>>> 
>>> kwargs
{'foo': 'bar'}
```
This makes it also easier to test the serialize function.

Reply:
@Fokko Thank you for taking time to review this PR, I've updated it based on your suggestions. 

```python
    def _write_args(self, input_filename):
        # serialize args to file
        if self.use_dill:
            serializer = dill
        else:
            serializer = pickle
        # some items from context can't be loaded in virtual env
        # see pr https://github.com/apache/airflow/pull/8256
        not_serializable = {'dag', 'task', 'ti', 'macros', 'task_instance', 'var'}
        if self._pass_op_args():
            kwargs = {key: value for key, value in self.op_kwargs.items()
                      if key not in not_serializable}
            with open(input_filename, 'wb') as f:
                arg_dict = ({'args': self.op_args, 'kwargs': kwargs})
                serializer.dump(arg_dict, f)
````

Reply:
Nice, thanks for taking the time to sort this out